config/crd/bases/pulpproject_v1beta1_pulp_crd.yaml

```diff
@@ -260,7 +260,10 @@ spec:
                   - Always
                   - Never
               image_pull_secret:
-                description: The image pull secret
+                description: Name of the secret that will be created with the content of image_pull_secret_credentials
+                type: string
+              image_pull_secret_credentials:
```

Comment:
@fao89  thank you for reviewing it :smile: 

> "we want to have a similar experience with awx-operator:"

How similar experience are we trying to achieve?
I'm asking this because this proposal implementation is similar in a way that it will provide a `secret` to
the controllers (`deployment/sts`) as an `imagePullSecret`, but instead of using an array of `secrets`, this commit
will use a single `secret` which can have **multiple credentials** in it (in the end, having the same behavior).

Another thing to notice is that `aws-operator` expects that users will [manually create the secret](https://github.com/ansible/awx-operator/blob/5e81729bc9586f0d7bca7ea4918ef78b050b2926/README.md#control-plane-ee-from-private-registry),
while this commit provides a new field (`image_pull_secret_credentials`) on `pulp CR` that will be used to "automatically" create the secret with it.
So, instead of creating the `secret(s)` through an ansible task, should we hold ansible execution until the user provides the secret, or should we let the playbook fails execution if the `secret(s)` was(were) not previously created?

Please, let me know if I understand it right.

Reply:
@rooftopcellist wdyt?

Reply:
If we adopted this approach, the user would be required to paste their pull secret credentials in plain text or base64 encoded in the pulp spec, which is undesirable.  In k8s, the main pattern for handling sensitive information is to use secrets. 

Consider the following scenario:
* customer creates a service account as an audit user, grants them permissions for all rules (get, list, describe, etc.) on all objects except secrets.  This way the audit user can see how a given deployment is used/setup, but not be able to see secret values. 

I don't know how many people use the "Encrypting Secret Data at Rest" feature of k8s, but this would probably be problematic for that too (https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/)